CesiumAsync/include/CesiumAsync/Impl/AsyncSystemSchedulers.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "CesiumAsync/Impl/cesium-async++.h"
+
+namespace CesiumAsync {
+
+class ITaskProcessor;
+
+namespace Impl {
+// Begin omitting doxgen warnings for Impl namespace
+//! @cond Doxygen_Suppress
+
+class AsyncSystemSchedulers;
+
+class ImmediatelyInMainThreadScheduler {
+public:
+  ImmediatelyInMainThreadScheduler(AsyncSystemSchedulers* pSchedulers)
+      : _pSchedulers(pSchedulers) {}
```

Comment:
since `ImmediatelyInMainThreadScheduler` uses `async::fifo_scheduler` in its method, maybe we should have this class own the `async::fifo_scheduler` directly and remove `AsyncSystemSchedulers` from it. That way, we won't have dependency between it and `AsyncSystemSchedulers`

Reply:
I'm not quite sure what the advantage of that would be. `ImmediatelyInMainThreadScheduler` is an embedded member of AsyncSystemSchedulers, so the lifetime dependency still exists. And the current approach is more symmetric with the other schedulers. Also, originally, `mainThreadScheduler` was meant to be usable directly, e.g. if you _don't_ want to execute immediately even if you're in the main thread. I eventually got rid of that because it wasn't immediately useful. But that's how JS promises work (by design) so it's easy to imagine we might want to support that behavior in the future.

Reply:
My initial thought is that with the proposed change, the `AsyncSystemSchedulers` still depends on the `ImmediatelyInMainThreadScheduler`, but if we can make the `ImmediatelyInMainThreadScheduler` not depend on the `AsyncSystemSchedulers` (even though it's harmless), I think we should